.github/workflows/refresh-lockfiles.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow periodically creates new environment lock files based on the newest
+# available packages and dependencies.
+#
+# Environment specifications are given as conda environment.yml files found in
+# ``requirements/py**.yml``.  These state the packages required, the conda channels
+# that the packages will be pulled from, and any versions of packages that need to be
+# pinned at specific versions.
+#
+# For environments that have changed, a pull request will be made and submitted
+# to the master branch.
+
+name: Refresh Lockfiles
+
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run once a week on a Saturday night.
+    - cron: 1 0 * * 6
+
+
+jobs:
+
+  gen_lockfiles:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: install nox
+        run: |
+          source $CONDA/bin/activate base
+          conda install -y -c conda-forge nox pyyaml
+      - name: generate lockfiles
+        run: $CONDA/bin/nox --session update_lockfiles
+      - name: create pull request
+        uses: peter-evans/create-pull-request@052fc72b4198ba9fbc81b818c6e1859f747d49a8
```

Comment:
I was wondering if there are any security implications to pointing to someone elses GitHub repo. I'm guessing that this is more safe because its pointing to a specific commit, but I was wondering how far you had considered this.

Reply:
You're more on it than I was when I first used GHA's - I had it pointing to a tag rather than a commit! Pointing to a specific commit is a recognised way of alleviating this security risk, [see here](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions).